core/templates/domain/objects/NumberWithUnitsObjectFactorySpec.ts

```diff
@@ -176,5 +191,74 @@ describe('NumberWithUnitsObjectFactory', () => {
         'SyntaxError: In "kg / m^(2)", "^" must be ' +
         'followed by a floating-point number');
     });
+
+    it('should create currency units', () => {
+      const createCurrencyUnitsSpy = spyOn(
+        nwuof.unitsFactory, 'createCurrencyUnits');
+      nwuof.createCurrencyUnits();
+      expect(createCurrencyUnitsSpy).toHaveBeenCalled();
+    });
+
+    it('should create NumberWithUnits object from dict', () => {
+      let numberWithUnitsObject = {
+        type: 'dummy-type',
+        real: 1,
+        fraction: {
+          isNegative: false,
+          wholeNumber: 2,
+          numerator: 1,
+          denominator: 3
+        },
+        units: [
+          {
+            unit: 'Kg',
+            exponent: 2
+          }
+        ]
+      };
+
+      let createdNumberWithUnits = nwuof.fromDict(numberWithUnitsObject);
+      expect(createdNumberWithUnits.toDict()).toEqual(numberWithUnitsObject);
+    });
+
+    it('should convert list to math.js compatible string', () => {
```

Comment:
Should we separate the tests based on the 'type'?

Reply:
I don't think it is required, other tests in the file also follow the same pattern